umh-core/pkg/service/s6/lifecycle_force.go

```diff
@@ -0,0 +1,151 @@
+		s.logger.Warnf("Failed to kill supervisor processes: %v", err)
+		// Continue with cleanup even if supervisor killing fails
+	}
+
+	// Remove directories with timeout awareness
+	if err := s.removeDirectoryWithTimeout(ctx, artifacts.ServiceDir, fsService); err != nil {
+		s.logger.Warnf("Failed to remove service directory: %v", err)
+	}
+
+	if err := s.removeDirectoryWithTimeout(ctx, artifacts.LogDir, fsService); err != nil {
+		s.logger.Warnf("Failed to remove log directory: %v", err)
+	}
+
+	// Verify cleanup completed
+	serviceExists, _ := fsService.PathExists(ctx, artifacts.ServiceDir)
+	logExists, _ := fsService.PathExists(ctx, artifacts.LogDir)
+
+	if serviceExists || logExists {
+		return fmt.Errorf("force cleanup incomplete: service=%v, log=%v", serviceExists, logExists)
+	}
+
+	s.logger.Infof("Force cleanup completed for service: %s", filepath.Base(artifacts.ServiceDir))
+	return nil
+}
+
+// killSupervisors kills s6-supervise processes
+func (s *DefaultService) killSupervisors(ctx context.Context, artifacts *ServiceArtifacts, fsService filesystem.Service) error {
+	supervisePaths := []string{
+		filepath.Join(artifacts.ServiceDir, "supervise"),
+		filepath.Join(artifacts.ServiceDir, "log", "supervise"),
+	}
+
+	var lastErr error
+	for _, supervisePath := range supervisePaths {
+		pidFile := filepath.Join(supervisePath, "pid")
+		if data, err := fsService.ReadFile(ctx, pidFile); err == nil && len(data) > 0 {
+			if pidStr := strings.TrimSpace(string(data)); pidStr != "" {
+				// Try SIGTERM first for graceful shutdown
+				if _, err := fsService.ExecuteCommand(ctx, "kill", "-TERM", pidStr); err != nil {
+					s.logger.Debugf("Failed to send SIGTERM to supervisor process %s: %v", pidStr, err)
+					lastErr = err
+				} else {
+					s.logger.Debugf("Sent SIGTERM to supervisor process %s", pidStr)
+				}
+
+				// Wait for graceful shutdown with timeout
+				select {
+				case <-time.After(gracePeriodForTermination):
+					// Process didn't terminate gracefully, use SIGKILL
+					if _, err := fsService.ExecuteCommand(ctx, "kill", "-KILL", pidStr); err != nil {
+						s.logger.Debugf("Failed to send SIGKILL to supervisor process %s: %v", pidStr, err)
+						lastErr = err
+					} else {
+						s.logger.Debugf("Sent SIGKILL to supervisor process %s after %v grace period", pidStr, gracePeriodForTermination)
+					}
+				case <-ctx.Done():
+					// Context cancelled, exit early
+					return ctx.Err()
+				}
+			}
+		}
+	}
+
+	return lastErr
+}
+
+// removeDirectoryWithTimeout removes a directory with timeout awareness
+// this has a high context time and should ONLY be called during
+// force cleanup operations where thoroughness is prioritized over speed.
+func (s *DefaultService) removeDirectoryWithTimeout(ctx context.Context, path string, fsService filesystem.Service) error {
+	// Use a short timeout for chunk-based deletion
+	const chunkTimeout = 750 * time.Millisecond
+
+	chunkCtx, cancel := context.WithTimeout(ctx, chunkTimeout)
```

Comment:
Isn't the outer ctx way smaller then 750, effectivly preventing us from fully using it ?